tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -1126,11 +1126,13 @@ def setUp(self):
         # Topic creation happens here
         super().setUp()
 
-    @cluster(num_nodes=8,
-             log_allow_list=[
-                 r"failed to hydrate chunk.*Connection reset by peer",
-                 r"failed to hydrate chunk.*NotFound"
-             ])
+    @cluster(
+        num_nodes=8,
+        log_allow_list=[
+            r"failed to hydrate chunk.*Connection reset by peer",
+            r"failed to hydrate chunk.*NotFound",
+            r"Failed to query spillover manifests: cloud_storage::error_outcome:10",
```

Comment:
Yeah, I had the same question/concern.

Reply:
> I'm wondering if it's possible to catch these errors and demote the log to a warning or info level log if the start offset has moved above the queried offset

I had a PR to try this https://github.com/redpanda-data/redpanda/pull/18574 but I have closed it as the current code path tries to avoid special cases or log level demotion while handling these errors, see @nvartolomei PR https://github.com/redpanda-data/redpanda/pull/18097 https://github.com/redpanda-data/redpanda/pull/18097/commits/d1543eeb8c929eccb414da678445fa0981fca075 which seems the right thing to do to me. 

We had a special case to demote the log level there even before https://github.com/redpanda-data/redpanda/pull/18574, and it seems complicated to keep it up to date.

https://github.com/redpanda-data/redpanda/pull/18574#issuecomment-2118839906 for additional context.

Reply:
I'm a bit confused, if https://github.com/redpanda-data/redpanda/pull/18574 was closed because the race is fixed, do we still need this PR?

Reply:
#18574 was closed because it tried something that went against the direction in #18097 . Before either of these PRs, the code in question attempted to detect cases where a miss occurred due to GC etc and tried to log this at debug level. Otherwise we had warning level logs.

* in #18574 I tried to expand the condition for classifying an error as debug log so that if the entire log is truncated and kafka offset still cannot be calculated, that is not classified as warning and is logged at debug level
* in #18097 the mechanism to classify some query misses as debug level is removed. Now everything is logged at warn level to be more explicit in terms of logging.

I closed #18574 as it went against the direction the code had taken in #18097 , not because the race condition was fixed. But I would expect all query misses to be logged as warn with the current state of the code, this seemed to happen frequently in CI failures, so this PR added the log to ignore list, I expected it still to be required.

But it looks like the CI failures no longer appear possibly due to some other changes, so this PR can be closed for now.